site/guide/state.md

```diff
@@ -85,6 +85,19 @@ fn from_request(req: &'a Request<'r>) -> request::Outcome<T, ()> {
 
 [`Request::guard()`]: https://api.rocket.rs/rocket/struct.Request.html#method.guard
 
+### Request-Local State
+
+In addition to application-wide state, each `Request` has its own state container
```

Comment:
I would start this by saying: "While managed state is _global_ and available application-wide, request-local state is _local_ to a given request, carried along with the request, and dropped once the request is completed." And then continue by describing why these semantics are useful.